lib/net.c

```diff
@@ -42,17 +43,35 @@
 
 static struct net_config config;
 
-/*
- * XXX The secret key of the RSS hash must be random
- * in order to avoid hackers to know it.
- */
-uint8_t default_rss_key[GATEKEEPER_RSS_KEY_LEN] = {
-	0x6d, 0x5a, 0x56, 0xda, 0x25, 0x5b, 0x0e, 0xc2,
-	0x41, 0x67, 0x25, 0x3d, 0x43, 0xa3, 0x8f, 0xb0,
-	0xd0, 0xca, 0x2b, 0xcb, 0xae, 0x7b, 0x30, 0xb4,
-	0x77, 0xcb, 0x2d, 0xa3, 0x80, 0x30, 0xf2, 0x0c,
-	0x6a, 0x42, 0xb7, 0x3b, 0xbe, 0xac, 0x01, 0xfa,
-};
+uint8_t default_rss_key[GATEKEEPER_RSS_KEY_LEN];
+
+/* Function to randomize the default RSS key. */
+static void
+randomize_rss_key(void)
+{
+	uint8_t i, temp_key, temp_count;
+	uint8_t final_set_count = 0;
+	ssize_t number_of_bytes;
+
+	number_of_bytes = getrandom(default_rss_key, GATEKEEPER_RSS_KEY_LEN, 0);
+	if (number_of_bytes != 40) {
```

Comment:
Please use ``sizeof(default_rss_key)`` instead of ``40`` or ``GATEKEEPER_RSS_KEY_LEN``, otherwise the reader may be confused about its meaning. Instead of using the recursive call, I would recommend to write it as (break it into lines as needed) ``while ((number_of_bytes = getrandom(default_rss_key, sizeof(default_rss_key), 0)) != sizeof(default_rss_key));``